pkg/options/jsonnet/jsonnet_applier.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jsonnet
+
+import (
+	"errors"
+	"fmt"
+
+	bundle "github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/apis/bundle/v1alpha1"
+	"github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/converter"
+	"github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/options"
+	"github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/options/openapi"
+	"github.com/google/go-jsonnet"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type disableImports struct{}
+
+func (disableImports) Import(importedFrom, importPath string) (jsonnet.Contents, string, error) {
+	return jsonnet.Contents{}, "", errors.New("jsonnet imports are disabled")
+}
+
+type applier struct {
+	importer jsonnet.Importer
+}
+
+// NewApplier creates a new optionts applier instance.
+func NewApplier(importer jsonnet.Importer) options.Applier {
+	if importer == nil {
+		importer = disableImports{}
+	}
+	return &applier{importer}
+}
+
+func (m *applier) ApplyOptions(comp *bundle.Component, opts options.JSONOptions) (*bundle.Component, error) {
```

Comment:
make sure to do a DeepCopy on the component.

Reply:
Done